src/StringFields.jl

```diff
@@ -31,8 +31,13 @@ function StringField(data::Vector{UInt8})
 end
 
 function StringField(data::SubString)
-    part = data.string.data[1+data.offset:data.offset+nextind(data, data.endof)-1]
-    return StringField(part, 1:length(data))
+    stringvec = Vector{UInt8}(data.string)
+    idx = 1+data.offset:data.offset+nextind(data, data.endof)-1
+    return StringField(stringvec[idx], 1:endof(data))
```

Comment:
`length` is the number of characters in a string, which is totally different (for non-ASCII data) from the last index in the string.